### PR TITLE
mark Maven core dependencies as provided

### PR DESCRIPTION
**Pull request by hboutemy**
this will reduce a little bit downloads when using the plugin

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -77,7 +77,7 @@
 
     <!-- Versions -->
     <java.level>1.8</java.level>
-    <maven.version>3.2.5</maven.version>
+    <maven.version>3.0.5</maven.version>
```
Comment (nicoulaj) [inline pom.xml:80]: reverting this back to 3.2.5, see discussion in #119

File: pom.xml [hunk 2]
```diff
@@ -99,16 +99,43 @@
...
         <version>${maven.version}</version>
+        <scope>provided</scope>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.maven</groupId>
+        <artifactId>maven-artifact</artifactId>
+        <version>${maven.version}</version>
+        <scope>provided</scope>
+      </dependency>
+      <dependency>
```
Comment (nicoulaj) [inline pom.xml:122]: @hboutemy can you explain why we need these dependencies ?

File: pom.xml (hunk 1 above)
Comment (hboutemy) [reply pom.xml:80]: oh yes, of course, sorry, I overlooked the mix :)

File: pom.xml (hunk 2 above)
Comment (hboutemy) [reply pom.xml:122]: great question
given Maven dependency conflict resolution algorithm, the dependencies I added were not automatically found from provided transitives but as transitives from compile scope: adding explicitely as dependencyManagement marks them provided as I wanted in this PR
notice: seeing that type of fact is one really nice feature of m2e in Eclipse

**Pull request merged**